### ISSUE TITLE
shellcheck thinks variable could expand to "/" when it can't

### ISSUE THREAD

**Issue by marcone** [koalaman/shellcheck#2167]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2115
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
mytmp=".mytempdir"
rm -rf "$BASEDIR/$mytmp"

```

#### Here's what shellcheck currently says:
```
Line 3	SC2115: Use "${var:?}" to ensure this never expands to / .
```
#### Here's what I wanted or expected to see:

Since "mytmp" is defined right there on line 2, shellcheck should know the expansion it complains about can't happen.
(to be clear: it's not complaining about BASEDIR. It wants me to write: `rm -rf "$BASEDIR/${mytmp:?}`)

**Comment by koalaman** [comment]
Through the power of bash, all things are possible:

```
/tmp # cat myscript
#!/bin/bash
mytmp=".mytempdir"
rm -rf "$BASEDIR/$mytmp"

/tmp # ./myscript
./myscript: line 2: mytmp: readonly variable
rm: can't remove '/sys/kernel/notes': Read-only file system
rm: can't remove '/sys/kernel/mm/page_idle/bitmap': Read-only file system
rm: can't remove '/sys/kernel/mm/hugepages/hugepages-2048kB/free_hugepages': Read-only file system
[...]
```

Here `mytmp` happened to be declared readonly in a BASH_ENV file, decimating an Alpine Linux image where `rm` doesn't have a safeguard against recursive operations on `/`. 

It's cheap insurance, but I agree that this is rather contrived and potentially bad refactoring without rerunning ShellCheck isn't a strong argument. If ShellCheck's dataflow analysis was more robust, it would at least be an optional or separate warning that could easily be suppressed.
